packages/spotlight/Spottable/useSpot.js

```diff
@@ -0,0 +1,97 @@
+ * @memberof ui/Spottable
+ * @property {Boolean}  disabled             Whether or not the component is in a disabled state.
+ * @property {Boolean}  emulateMouse         Whether or not the component should emulate mouse events as a response to Spotlight 5-way events.
+ * @property {Function} onMouseUp            Is is called if the component is focused and became disabled.
+ * @property {Function} onSpotlightDisappear The handler to run when the component is removed while retaining focus.
+ * @property {Function} onSpotlightDown      The handler to run when the 5-way down key is pressed.
+ * @property {Function} onSpotlightLeft      The handler to run when the 5-way left key is pressed.
+ * @property {Function} onSpotlightRight     The handler to run when the 5-way right key is pressed.
+ * @property {Function} onSpotlightUp        The handler to run when the 5-way up key is pressed.
+ * @property {Number[]} selectionKeys        An array of numbers representing keyCodes that should trigger mouse event
+ *                                           emulation when `emulateMouse` is `true`. If a keyCode equals a directional
+ *                                           key, then default 5-way navigation will be prevented when that key is pressed.
+ * @property {Boolean}  spotlightDisabled    When `true`, the component cannot be navigated using spotlight.
+ * @private
+ */
+
+/**
+ * Object returned by `useSpot`
+ *
+ * @typedef {Object} useSpotInterface
+ * @memberof ui/Spottable
+ * @property {Boolean}  [blur]       Handle when blurred.
+ * @property {Boolean}  [className]  The class being spottable and focused when disabled.
+ * @property {Boolean}  [focus]      Handle when focused.
+ * @property {Boolean}  [keyDown]    Handle to run when the 5-way up key is pressed.
+ * @property {Boolean}  [keyUp]      Handle to run when the 5-way up key is released.
+ * @property {Boolean}  [mouseEnter] Handle when mouse enters.
+ * @property {Boolean}  [mouseLeave] Handle when mouse leaves.
+ * @property {Boolean}  [ref]        The ref for the target node.
+ * @private
+ */
+
+/**
+ * Manages a boolean state value.
+ *
+ * The value may either be Spotd via the `onSpot` or explicitly set via `onActivate` and
+ * `onDeactivate`. The initial value can be set using the `defaultSelected` option.
+ *
+ * @param {useSpotConfig} config Configuration options
+ * @returns {useSpotInterface}
+ * @private
+ */
+
+const useSpot = ({selectionKeys = [ENTER_KEY, REMOTE_OK_KEY], spotlightDisabled, ...config} = {}) => {
+	const useForceUpdate = () => (React.useReducer(x => x + 1, 0));
+	const spot = useClass(Spot, {selectionKeys, spotlightDisabled, useForceUpdate, ...config});
+	const ref = React.useRef(null);
+	const mutableRef = React.useRef({
+		prevSpotlightDisabled: spotlightDisabled,
+		spotlightDisabled
+	});
+
+	spot.setContext(mutableRef);
+
+	React.useEffect(() => {
+		// eslint-disable-next-line react/no-find-dom-node
+		spot.load(ReactDOM.findDOMNode(ref.current));
```

Comment:
Hello @ryanjduffy 

I had one problem to migrate Spottable to use Hooks.
In detail, I don't know how to change the `ReactDOM.findDOMNode(this)` properly on the `useSpot`. With the PR, there is the following warning. Do we have to keep using the `Spottable` as the React Component?

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
      
      Check the render method of `Spottable`.
          in mockConstructor (at Spottable.js:148)
          in Spottable (created by WrapperComponent)
          in WrapperComponent
```

Reply:
In my opinion, there is no problem that the `Spottable` works. And there are no errors to run lint in enact and sandstone. So could we go ahead and fix it later? Or should we have to fix it in this PR?

Reply:
To maintain `ref` properly, I defined `Spottable` based on React.Component and passed the `componentRef` from `ReactDOM.findDOMNode(this)` to `SpottableBase` and `useSpot`.

```
	class Spottable extends React.Component {
		componentDidMount () {
			// eslint-disable-next-line react/no-find-dom-node
			this.node = ReactDOM.findDOMNode(this);
		}

		get componentRef () {
			return this.node;
		}

		render () {
			return <SpottableBase {...this.props} componentRef={this.componentRef} />;
		}
	}
```